src/flexmeasures_client/response_handling.py

```diff
@@ -26,10 +26,12 @@ async def check_response(self: FlexMeasuresClient, response, polling_step: int):
     print(status)
     if status < 300:
         pass
-    elif status == 401:
+    elif status == 401 and payload.get("status") == "UNAUTHORIZED":
```

Comment:
I was expecting this to be the fix:
```
elif  status == 401 and self.reauth_once:
```
Why the check for UNAUTHORIZED instead?

Reply:
That solution does not work. `self.reauth` is set to true in `request`. So once it tries to execute the request again it will try to reauthenticate. 

UNAUTHORIZED is the status response if the wrong `self.access_token` is used for a request, triggering a reauth which can not fail in the same way because `get_access_token()` doesn't use the access_token.

Reply:
It seems like a rather implicit way of solving the issue. The goal is to prevent get_access_token from triggering itself in a loop, right?

An alternative could be to only attempt to reauthenticate if the auth header is used. This could be done by setting reauth_once based on include_auth. In fact, we may just as well use include_auth instead of reauth_once to decide whether to attempt to reauthenticate. And that would also mean we can get rid of setting up reauth_once as a class parameter.

Reply:
I still found the UNAUTHORIZED check too implicit, which worried me that future developers will find it hard to know what this is for. I also found the `self.headers` approach risky with possible the wrong headers used when different methods are called simultaneously, and multiple polling loops ongoing. I found the trick was to move the `get_headers` into the polling loop and actually connecting the incremented polling step, which had been disconnected in `request_once`.

The only thing that doesn't look too elegant now is the passing back and forth of the `polling_step` and the `reauth_once` parameters. Perhaps at some point we can create a local request_context variable or something, where we can store this info, and update them inplace. But that's for another time.

To summarize, I feel this new approach is not too elegant, but more explicit, and therefore has my preference.